diff --git a/third_party/protobuf-lite/google/protobuf/stubs/hash.h b/third_party/protobuf-lite/google/protobuf/stubs/hash.h
--- a/third_party/protobuf-lite/google/protobuf/stubs/hash.h
+++ b/third_party/protobuf-lite/google/protobuf/stubs/hash.h
@@ -235,9 +235,9 @@ class hash_set : public std::set<Key, HashFcn> {
   HashFcn hash_function() const { return HashFcn(); }
 };
 
-#elif defined(_MSC_VER) && !defined(_STLPORT_VERSION) && \
+#elif defined(_MSC_VER) && _MSC_VER < 1600 && !defined(_STLPORT_VERSION) && \
     !(defined(_LIBCPP_STD_VER) && _LIBCPP_STD_VER >= 11)
 
 template <typename Key>
 struct hash : public GOOGLE_PROTOBUF_HASH_COMPARE<Key> {
