diff --git a/cmake/FindPCRE2.cmake b/cmake/FindPCRE2.cmake
--- a/cmake/FindPCRE2.cmake
+++ b/cmake/FindPCRE2.cmake
@@ -45,9 +45,10 @@
   set(PCRE2_CONFIGURE_OPTIONS
     -DBUILD_SHARED_LIBS=OFF
     -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}
     -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
     ${PCRE2_JIT_OPTION}
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} # Necessary for proper MacOS compilation
     -DCMAKE_CROSSCOMPILING_EMULATOR=${CMAKE_CROSSCOMPILING_EMULATOR_WITH_SEMICOLON} # Necessary for proper MacOS compilation
     -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true # Added for pybind11
@@ -56,5 +57,9 @@
   # set include dirs and libraries for PCRE2
-  set(PCRE2_LIBRARIES ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}pcre2-8${CMAKE_STATIC_LIBRARY_SUFFIX})
+  if(WIN32)
+    set(PCRE2_LIBRARIES ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}pcre2-8-static${CMAKE_STATIC_LIBRARY_SUFFIX})
+  else()
+    set(PCRE2_LIBRARIES ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}pcre2-8${CMAKE_STATIC_LIBRARY_SUFFIX})
+  endif()
   set(PCRE2_INCLUDE_DIRS "${CMAKE_BINARY_DIR}/include")
   set(PCRE2_FOUND TRUE CACHE BOOL "Found PCRE2 libraries" FORCE)
   
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,8 @@
 include_directories(ssplit)
 add_library(ssplit STATIC ssplit/ssplit.cpp ssplit/regex.cpp)
+if(WIN32 AND SSPLIT_USE_INTERNAL_PCRE2)
+  target_compile_definitions(ssplit PRIVATE PCRE2_STATIC)
+endif()
 target_link_libraries(ssplit ${EXT_LIBS})
 
 set_target_properties(ssplit PROPERTIES
